### PR TITLE
Enable codec2 en/decoding using libcodec2

### PR DESCRIPTION
**Pull request by schmidthubert**
Enable codec2 en/decoding using libcodec2 as per recent commit https://git.ffmpeg.org/gitweb/ffmpeg.git/commitdiff/f958f431eced497f42220d8f9890506215742512

### CODE REVIEW

**Comment by schmidthubert** [comment]
--enable-codec2

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -582,6 +582,23 @@ if [[ $ffmpeg != "no" ]] && enabled libsoxr; then
...
 
+if [[ $ffmpeg != "no" ]] && enabled libcodec2 && do_pkgConfig "codec2 = 0.7"; then 
+    _check=(libcodec2.a codec2.pc) 
+	if do_wget -h 0695bb93cd985dd39f02f0db35ebc28a98b9b88747318f90774aba5f374eadb2 \
+            "https://freedv.com/wp-content/uploads/sites/8/2017/10/codec2-0.7.tar.xz"; then
+        do_uninstall include/codec2 "${_check[@]}"
+		touch cmake/GetPrerequisites.cmake
+        sed -i 's|include(${CMAKE_SOURCE_DIR}/cmake/GetPrerequisites.cmake)||' cmake/GetDependencies.cmake.in
+        sed -i 's|get_prerequisites(${CODEC2_DLL} _deps 1 0 "" "")||' cmake/GetDependencies.cmake.in
+		do_cmake -G "MSYS Makefiles"
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: This shouldn't work, since do_cmake already sets `-G Ninja`.

File: build/media-suite_compile.sh [hunk 2]
```diff
@@ -582,6 +582,23 @@ if [[ $ffmpeg != "no" ]] && enabled libsoxr; then
     fi
 fi
 
+if [[ $ffmpeg != "no" ]] && enabled libcodec2 && do_pkgConfig "codec2 = 0.7"; then 
+    _check=(libcodec2.a codec2.pc) 
+	if do_wget -h 0695bb93cd985dd39f02f0db35ebc28a98b9b88747318f90774aba5f374eadb2 \
+            "https://freedv.com/wp-content/uploads/sites/8/2017/10/codec2-0.7.tar.xz"; then
+        do_uninstall include/codec2 "${_check[@]}"
+		touch cmake/GetPrerequisites.cmake
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: Normalize these tabs/spaces. 4 space indent only for each indentation level.

File: build/media-suite_compile.sh [hunk 3]
```diff
@@ -582,6 +582,23 @@ if [[ $ffmpeg != "no" ]] && enabled libsoxr; then
...
+        sed -i 's|include(${CMAKE_SOURCE_DIR}/cmake/GetPrerequisites.cmake)||' cmake/GetDependencies.cmake.in
+        sed -i 's|get_prerequisites(${CODEC2_DLL} _deps 1 0 "" "")||' cmake/GetDependencies.cmake.in
+		do_cmake -G "MSYS Makefiles"
+        do_makeinstall
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: Unless this doesn't work with ninja, replace this with
```
log make ninja
log install ninja install
```

Comment (schmidthubert) [reply build/media-suite_compile.sh]: Doesn't work for me at all: 
ninja: error: loading 'build.ninja': Das System kann die angegebene Datei nicht finden.
Which means 'build.ninja' cannot be found. 

I works as per my patch in 32bit and 64bit, tested.

File: build/media-suite_compile.sh (hunk 2 above)
Comment (schmidthubert) [reply build/media-suite_compile.sh]: I'm just using tabs in Notepad++ as always but anyways ....

File: build/media-suite_compile.sh (hunk 1 above)
Comment (schmidthubert) [reply build/media-suite_compile.sh]: What do you mean? The same command is used on line 1747 in libneon (for example)???

Comment (wiiaboo) [reply build/media-suite_compile.sh]: Yeah, nevermind, it should work.

File: build/media-suite_compile.sh (hunk 3 above)
Comment (wiiaboo) [reply build/media-suite_compile.sh]: You'd need to remove the `-G "MSYS Makefiles"` first so the ninja files are generated.
If ninja works, it's way better than using make.

**Comment by schmidthubert** [comment]
Tested ninja, working now. Dpn't know where to put --enable-libcodec2 in the batch (if at all?). I have it in my ffmpeg_options.txt ...

**Comment by wiiaboo** [comment]
You don't necessarily need to add it to the defaults for it to work, but 0bedbea is fine. Add it to the readme as well, for consistency.

**Comment by schmidthubert** [comment]
>Add it to the readme as well, for consistency.
Done.

**Comment by wiiaboo** [comment]
Great, I'll just try it once and squash and merge it.

**Comment by wiiaboo** [comment]
This builds and installs a lot of binaries to /bin. Are all of them needed?

**Comment by schmidthubert** [comment]
Sorry, I don't know that Ricardo! The codec2 sounds like complicated stuff so I guess every file has their purpose.

File: build/media-suite_compile.sh [hunk 4]
```diff
@@ -439,7 +439,7 @@ if [[ $ffmpeg != "no" ]] && enabled libilbc && do_pkgConfig "libilbc = 2.0.3-dev
 fi
 
 enabled libvorbis && do_pacman_install libvorbis
-enabled libspeex && do_pacman_install speex
+enabled_any libspeex libcodec2 && do_pacman_install speex
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: speex is only needed for UNITTEST=on. You don't need this change.

File: build/media-suite_compile.sh [hunk 5]
```diff
@@ -582,6 +582,24 @@ if [[ $ffmpeg != "no" ]] && enabled libsoxr; then
...
+    _check=(libcodec2.a codec2.pc) 
+	if do_wget -h 0695bb93cd985dd39f02f0db35ebc28a98b9b88747318f90774aba5f374eadb2 \
+            "https://freedv.com/wp-content/uploads/sites/8/2017/10/codec2-0.7.tar.xz"; then
+        do_uninstall include/codec2 "${_check[@]}"
+        touch cmake/GetPrerequisites.cmake
+        sed -i 's|include(${CMAKE_SOURCE_DIR}/cmake/GetPrerequisites.cmake)||' cmake/GetDependencies.cmake.in
+        sed -i 's|get_prerequisites(${CODEC2_DLL} _deps 1 0 "" "")||' cmake/GetDependencies.cmake.in
+        do_cmake
+        log make ninja
+        log install ninja install
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: I'd rather disable everything that's binaries for standalone=y. In this case it's probably easier to manually install the headers/lib/pkgconfig and then manually install c2{dec,enc,sim}.exe to bin-audio if standalone=y. If someone eventually needs other binaries, install them later when asked.

Comment (schmidthubert) [reply build/media-suite_compile.sh]: Done.

**Comment by wiiaboo** [comment]
Merged to 1d39ef92429ac99b271772e8257d214039a86719, thanks.

**Comment by schmidthubert** [comment]
Thanks! :+1:

**Pull request closed without merging**